WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -39,14 +39,14 @@ def psych_sheet_event
     @event = Event.find(params[:event_id])
     @preferred_format = @event.preferred_formats.first
 
-    # TODO - pull registered events out into a join table
-    # https://github.com/cubing/worldcubeassociation.org/issues/275#issuecomment-167347053
-    @registrations = @competition.registrations.accepted.all.select { |r|
-      r.events.include?(@event)
-    }.sort_by { |r|
-      has_competed = !!r.world_rank(@event, @preferred_format.sort_by)
-      [ has_competed ? 0 : 1, r.world_rank(@event, @preferred_format.sort_by) || Float::INFINITY, r.world_rank(@event, @preferred_format.sort_by_second) || Float::INFINITY, r.name ]
-    }
+    @registrations = @competition.registrations.
+                                  accepted.
+                                  joins(:registration_events).
+                                  where("registration_events.event_id=?", @event.id).
+                                  sort_by do |r|
+                                    has_competed = !!r.world_rank(@event, @preferred_format.sort_by)
```

Comment:
What about adding a method `has_competed_in?(event)` to the user model? Btw. I think that I saw such a line somewhere else, so probably we would be able to reuse it even now =)